etcd/requests.go

```diff
@@ -303,8 +303,8 @@ func DefaultCheckRetry(cluster *Cluster, numReqs int, lastResp http.Response,
 	err error) error {
 
 	if numReqs > 2*len(cluster.Machines) {
-		return newError(ErrCodeEtcdNotReachable,
-			"Tried to connect to each peer twice and failed", 0)
+		errStr := fmt.Sprintf("Tried to connect to members %v twice and failed: %v", cluster.Machines, err)
```

Comment:
The error that brought me to this bug was "413 too large", which does not warrant a retry, and which your client will now report in the error string but will otherwise totally obfuscate.  So my assessment is still "better but barely".  This is such an important client lib it should really handle errors better, and that's not a 1-liner to fix.